components/availability/src/Availability.svelte

```diff
@@ -536,9 +542,41 @@
     timestamp: Date;
   }
 
+  function checkOverbooked(slots: SelectableSlot[]) {
+    allCalendars.forEach((calendar) => {
+      let availableSlotsForCalendar = slots.filter((slot) =>
+        slot.available_calendars.includes(calendar.account?.emailAddress),
+      );
+      if (
+        availableSlotsForCalendar.length >
+        (overbooked_threshold * slots.length) / 100
+      ) {
+        availableSlotsForCalendar.forEach((slot) => {
+          slot.available_calendars = slot.available_calendars.filter(
+            (cal) => cal !== calendar.account?.emailAddress,
+          );
+          if (!slot.available_calendars.length) {
+            // if it has no calendars available, it's busy
+            slot.availability = AvailabilityStatus.BUSY;
+          } else if (
+            slot.availability === AvailabilityStatus.FREE &&
+            slot.available_calendars.length !== allCalendars.length
+          ) {
+            // if it was previously free, but now lacks a calendar, it should be considered Partial.
+            slot.availability = AvailabilityStatus.PARTIAL;
+          }
+        });
+      }
+    });
+
+    return slots;
+  }
+
   let days: Day[];
   $: days = dayRange.map((timestamp: Date) => {
-    let slots = generateDaySlots(timestamp, start_hour, end_hour);
+    let slots = checkOverbooked(
+      generateDaySlots(timestamp, start_hour, end_hour),
+    ); // TODO: include other potential post-all-slots-established checks, like overbooked, in a single secondary run here.
```

Comment:
we talked about refactoring htis but as a part of the other refactor going on this week